common/lib/dependabot/shared_helpers.rb

```diff
@@ -374,12 +374,35 @@ def self.configure_git_to_use_https(host)
 
     sig { params(path: String).void }
     def self.reset_git_repo(path)
-      Dir.chdir(path) do
-        run_shell_command("git reset HEAD --hard")
-        run_shell_command("git clean -fx")
+      if lfs_enabled?(path)
+        Dir.chdir(path) do
+          begin
+            run_shell_command("git-lfs-reset HEAD --hard")
+          rescue SharedHelpers::HelperSubprocessFailed
+            Dependabot.logger.warn("LFS is enabled in this repo.  Please use an LFS enabled client")
+          end
+          run_shell_command("git clean -fx")
+        end
+      else
+        Dir.chdir(path) do
+          run_shell_command("git reset HEAD --hard")
+          run_shell_command("git clean -fx")
+        end
       end
     end
 
+    sig { params(path: String).returns(T.nilable(T::Boolean)) }
+    def self.lfs_enabled?(path)
+      filepath = File.join(path, ".gitattributes")
+      T.let(true, T::Boolean) if File.exist?(filepath) && File.readable?(filepath) &&
+                                        SharedHelpers.run_shell_command("cat #{filepath} | grep \"filter=lfs\"")
+                                                     .include?("filter=lfs")
+    rescue StandardError => e
```

Comment:
Jurre's [suggestion](https://github.com/dependabot/dependabot-core/pull/9766#discussion_r1620398131) is more idiomatic and won't incur the cost of shelling out, so I think you should incorporate that.

Just to call attention to it, shelling out to grep like this will return a non-zero exit if there's no match. `SharedHelpers.run_shell_command` will [raise](https://github.com/dependabot/dependabot-core/blob/87d4dfe741e2297d65776a5825dfe40bf31f460d/common/lib/dependabot/shared_helpers.rb#L421-L435) a `HelperSubprocessFailed` exception on a non-zero exit, but here we're swallowing virtually _any_ error that could occur and carrying on. The problem with this is it assumes the only thing that could go wrong is grep not finding a match, when there are other legitimate exceptions that could occur that we would want to propagate.

If you find yourself writing `rescue StandardError`, it might be a good idea to ask if it's really necessary for what you're trying to do. If you truly want to check for the `HelperSubprocessFailed` exception, rescue that class specifically so that other, unexpected exceptions are propagated correctly.